crates/bevy_math/src/cubic_splines.rs

```diff
@@ -1205,10 +1251,14 @@ pub trait RationalGenerator<P: VectorSpace> {
 }
 
 /// A segment of a rational cubic curve, used to hold precomputed coefficients for fast interpolation.
-/// Can be evaluated as a parametric curve over the domain `[0, knot_span)`.
+/// It is a [`Curve`] with domain `[0, 1]`.
```

Comment:
Are you sure about this?
If this is true, there's probably a bunch of wrong comments down the line in the impl.

Reply:
Yeah. If you look at the implementation of `RationalCurve::segment`, you'll see that the `t`-value used in deferring to `RationalSegment` for sampling is normalized to `[0, 1]`. The only error here is that I neglected to fix the other doc comments in `RationalSegment` to explain that it's `[0, 1]` and not `[0, knot_span]`.

(`knot_span` is used by `RationalCurve` to perform that normalization in the first place.)

Reply:
It looks like I also forgot to change the domain of `RationalSegment` as a curve to `Interval::UNIT`. I guess there's a question of whether we should make it work like its `knot_span` indicates it might, but
1. I don't think that's especially important to resolve now, and
2. Using `Interval::UNIT` lines up with how the existing sampling methods on `RationalSegment` are actually implemented

Reply:
Alright, thanks for catching this.